src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModel.java

```diff
@@ -62,12 +62,46 @@ public static WeeklyHabitScheduleModel parseList(List<String> list) {
 
     @Override
     public boolean isToBeCompletedOn(Date date) {
-        return false; // TODO
+        Date today = new Date();
+        int testedDateOfWeek = date.getDay();
+        if (date.after(today)) {
+            return days[testedDateOfWeek];
+        }
+        return false;
     }
 
     @Override
     public boolean wasSkippedIfLastCompletedOn(Date date) {
-        return false; // TODO
+        int lastCompletedDateOfWeek = date.getDay();
```

Comment:
a few minor styling issues in this function, try selecting the function contents and running Ctrl+Alt+L